src/Symfony/Component/DependencyInjection/Compiler/CheckDefinitionValidityPass.php

```diff
@@ -65,6 +65,14 @@ public function process(ContainerBuilder $container)
                         $id
                     ));
                 }
+                if (preg_match('/^\\\\[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+(?:\\\\[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+)++$/', $id)) {
+                    throw new RuntimeException(sprintf(
+                        'The definition for "%s" has no class attribute, and appears to reference a class or interface. '
```

Comment:
> and appears to reference a class or interface

actually not, since the class doesn't exist
that + the fact that the message reported in #33092 is not the one just below, I'm not sure this PR is the correct answer.

Reply:
you're right. Let's continue in #33108 

AFAIK the if block at L51 is now dead code :/